src/NextGen/identity/identity.ts

```diff
@@ -1,48 +1,134 @@
+    | 'twitter'
+    | 'google'
+    | 'microsoft'
+    | 'yahoo'
+    | 'email'
+    | 'facebookcustomaudienceid'
+    | 'other2'
+    | 'other3'
+    | 'other4'
+    | 'other5'
+    | 'other6'
+    | 'other7'
+    | 'other8'
+    | 'other9'
+    | 'other10'
+    | 'phone_number_2'
+    | 'phone_number_3'
+    | 'mobile_number'
+    | 'device_application_stamp';
+
+type KnownIdentities = {
+    [key in IdentityTypes]?: string;
+};
+
+export interface IdentityApiRequest {
+    client_sdk: {
+        platform: string;
+        sdk_vendor: string;
+        sdk_version: string;
+    };
+    context: Context;
+    environment: 'development' | 'production';
+    request_id: string;
+    request_timestamp_ms: number;
+    previous_mpid: MPID | null;
+    known_identities: KnownIdentities;
+}
 
 export interface IMParticleIdentity extends IMParticleComponent {
-    login(request: IdentityApiRequest): Promise<IdentityApiResult>;
-    logout(request: IdentityApiRequest): Promise<IdentityApiResult>;
+    login(request: IdentityApiRequest): Promise<IMPIdentityApiResponse>;
+    logout(request: IdentityApiRequest): Promise<IMPIdentityApiResponse>;
     getUser(mpid: MPID): Promise<User>;
     getUsers(): Promise<User[]>;
 }
 
 export interface InternalIdentity extends IMParticleIdentity {
-    login(request: IdentityApiRequest): Promise<IdentityApiResult>;
-    logout(request: IdentityApiRequest): Promise<IdentityApiResult>;
-    modify(request: IdentityApiRequest): Promise<IdentityApiResult>;
-    identify(request: IdentityApiRequest): Promise<IdentityApiResult>;
+    login(request: IdentityApiRequest): Promise<IMPIdentityApiResponse>;
+    logout(request: IdentityApiRequest): Promise<IMPIdentityApiResponse>;
+    modify(request: IdentityApiRequest): Promise<IMPIdentityApiResponse>;
+    identify(request: IdentityApiRequest): Promise<IMPIdentityApiResponse>;
 }
 
 export interface IdentityListener {}
 
 export class IdentityImplementation implements InternalIdentity {
     constructor() {}
 
-    public identify(request: IdentityApiRequest): Promise<IdentityApiResult> {
-        return Promise.resolve({});
+    // TODO: Temporary implementation to test signature
+    private generateFakeAPIResult = () => ({
+        user: null,
+        previousUser: null,
+        getUser: () => {
+            return null;
+        },
+        getPreviousUser: () => {
+            return null;
+        },
+    });
+
+    public identify(
+        request: IdentityApiRequest
+    ): Promise<IMPIdentityApiResponse> {
+        return Promise.resolve(this.generateFakeAPIResult());
     }
 
-    public modify(request: IdentityApiRequest): Promise<IdentityApiResult> {
-        return Promise.resolve({});
+    public modify(
+        request: IdentityApiRequest
+    ): Promise<IMPIdentityApiResponse> {
+        // return Promise.resolve(this.generateFakeAPIResult());
+        return Promise.reject({
+            errorCode: MParticleServerErrorCodes.activeSession,
+        });
```

Comment:
should line 118 be uncommented and this be refactored to not have both reject/resolve?

Reply:
My intention was to hard code a single one of these methods to reject rather than resolve so we can visually see both use success and error cases. We can then remove the hardcoded reject/resolve once the real implementation is in place.